modin/pandas/groupby.py

```diff
@@ -658,8 +683,9 @@ def transform(self, func, *args, **kwargs):
         result = self._apply_agg_function(
             lambda df: df.transform(func, *args, **kwargs)
         )
-        # pandas does not name the index on transform
-        result._query_compiler.set_index_name(None)
+        if self._by is not None:
+            # pandas does not name the index on transform
```

Comment:
maybe we shouldn't use concrete names of the methods which pandas does not name the index on and use generic definition of such operations?

Reply:
I'm not sure that I got what you mean. Do you propose to change this comment line with deleting `transform` here?

Reply:
Discussed offline. It was solved to leave it as it is because operations which reset the index name are different